File: training/passwordprotecting/python/token.py
class UserToken:

  def __init__(self, userId:int, token:str) -> None:
    self._token = token
    self._userId = userId
    self._createdDateTime = 0
    self._isValid = True

  def userId(self):
    return self._userId
  
  def token(self):
    return self._token
  
  def invalidateToken(self):
    self._isValid = False

File: training/passwordprotecting/javascript/main/support/token.js
class UserToken {
  _token;
  _userId;
  _createdDateTime; // msec
  _isValid = true;

  /**
   * 
   * @param {int} userId 
   * @param {GUID} token 
   */
  constructor(userId, token) {
    this._token = token;
    this._userId = userId;
    this._createdDateTime = Date.now();
  }

  getUserId() {
    return this._userId;
  }

  getToken() {
    // let currentCutoff = Date.now() - expiration;
    return this._token;
  }

  invalidateToken() {
    this._isValid = false;
  }

}

export default UserToken;
